flocker/common/_interface.py

```diff
@@ -78,3 +80,50 @@ def invariant(value):
     invariant.__name__ = "provides_{}_invariant".format(interface_name)
 
     return invariant
+
+
+@attributes([
+    Attribute("unexpected_arguments", default_value=frozenset()),
+    Attribute("missing_arguments", default_value=frozenset()),
+    Attribute("missing_optional_arguments", default_value=frozenset())
+])
+class InvalidSignature(Exception):
+    """
+    See :func:`validate_signature_against_kwargs` raises clause.
+    """
+    def __str__(self):
+        return repr(self)
+
+
+def validate_signature_against_kwargs(function, keyword_arguments):
+    """
+    Validates that ``function`` can be called with keyword arguments with
+    exactly the passed in ``keyword_arguments_keys``. In this case validation
```

Comment:
:question: "the passed in" is rather clunky, and looks wrong to me until half way through. "with exactly the names in ``keyword_argument_keys`` or something sounds better to me, or "specified" instead of "passed in"